src/libworld/libworld.h

```diff
@@ -2541,6 +2546,36 @@ namespace world
         static chrono::milliseconds getLogTimestamp();
         static void formatLogString(chrono::milliseconds timestamp, char logString[512], const char* format, va_list args);
     };
+
+    class WorldRoutes
+    {
+        public:
+        class Route{
+            private:
+                string m_icaoDeparture;
+                string m_icaoDestination;
+                std::vector<string> m_icaoAirframes;
+                string m_AirlineCallsign;
+                string m_AirlineIcao;
+            public:
+                const string& departure()   const {return m_icaoDeparture;}
+                const string& destination() const {return m_icaoDestination;}
+                const std::vector<string>& usedAirframes() const {return m_icaoAirframes;}
+                const string& callsign()    const {return m_AirlineCallsign;}
+                const string& airline()    const {return m_AirlineIcao;}
+            public:
+                Route(const string _departure, const string _destination, const string _airlineIcao, const string _callsign, std::vector<string> _airframes):
+                    m_icaoDeparture(_departure),
+                    m_icaoDestination(_destination),
+                    m_icaoAirframes(_airframes),
+                    m_AirlineIcao(_airlineIcao),
+                    m_AirlineCallsign(_callsign)
+                    {}
+        };
+        public:
+        virtual const Route& findRandomRouteFrom(const string &fromICAO, const string &airframe, const std::vector<std::string>& allowedAirlines) = 0;
```

Comment:
LMK if you want me to rename **libdataxp** -> **libloaders** - I can merge this change into master, then you can update your branch.

Reply:
If you don't need the new name now, i'll do it.

Reply:
For the decoupling, I totally agree with you.
Maybe I should rename WorlRoutes as RouteProvider so we could use it as a base class for different services :

- Offline Airlines routes (this one base on OpenFlights data)
- Online Airlines routes (possibly multiple versions backed by different services)
- GA routes (Random destination within a range compatible with the aircraft) 

I'll keep the interface as is for now but maybe i should change it and provide an iterator instead of only one route.